tests/backup/backup_helper.go

```diff
@@ -5559,3 +5561,172 @@ func UpgradeKubevirt(versionToUpgrade string, workloadUpgrade bool) error {
 	}
 	return nil
 }
+
+// UpdateKDMPConfigMap update the KDMP configMap with give data key and value.
+func UpdateKDMPConfigMap(dataKey string, dataValue string) error {
+	KDMPconfigMap, err := core.Instance().GetConfigMap("kdmp-config", "kube-system")
+	if err != nil {
+		return err
+	}
+	KDMPconfigMap.Data[dataKey] = dataValue
+	_, err = core.Instance().UpdateConfigMap(KDMPconfigMap)
+	if err != nil {
+		return err
+	}
+	log.Infof("updated the KDMP configMap data with key [%s]: value [%s]", dataKey, dataValue)
+	return nil
+}
+
+// CreateNestedDirectoriesWithFilesInPod creates a nested directory structure with files within a specified Pod.
+func CreateNestedDirectoriesWithFilesInPod(podName, podNamespace, basePath string, depth, levels, filesPerDirectory int) error {
+	var errChan = make(chan error)
+	for i := 0; i < depth; i++ {
+		dirPath := filepath.Join(basePath, fmt.Sprintf("level_%d", i))
+		err := CreateDirectoryInPod(podName, podNamespace, dirPath)
+		if err != nil {
+			fmt.Printf("Error creating directory %s: %s\n", dirPath, err.Error())
+			continue
+		}
+		if levels > 1 {
+			wg.Add(1)
+			go func(dirPath string) {
+				defer wg.Done()
+				errChan <- CreateDirectoryStructureInPod(podName, podNamespace, dirPath, depth, levels-1, filesPerDirectory)
+			}(dirPath)
+		}
+
+	}
+	if filesPerDirectory != 0 {
+		err := CreateFilesInPodDirectory(podName, podNamespace, basePath, filesPerDirectory)
+		if err != nil {
+			return fmt.Errorf("error creating files in %s: %s", basePath, err.Error())
+		}
+	}
+	go func() {
+		wg.Wait()
+		close(errChan)
+	}()
```

Comment:
Why does this have to run in a goroutine? This can run at the end of the function right?